src/libraries/System.Private.CoreLib/src/System/Numerics/VectorMath.cs

```diff
@@ -11,68 +10,131 @@ namespace System.Numerics
+            }
+            else
+            {
+                // Redundant test so we won't prejit remainder of this method on platforms without AdvSimd.
+                throw new PlatformNotSupportedException();
+            }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Vector128<float> ConditionalSelectBitwise(Vector128<float> selector, Vector128<float> ifTrue, Vector128<float> ifFalse)
+        public static Vector128<float> Lerp(Vector128<float> a, Vector128<float> b, Vector128<float> t)
         {
-            Debug.Assert(Sse.IsSupported || AdvSimd.IsSupported);
+            // This implementation is based on the DirectX Math Library XMVectorLerp method
+            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
 
-            if (Sse.IsSupported)
+            if (AdvSimd.IsSupported)
             {
-                return Sse.Or(
-                            Sse.And(ifTrue, selector),
-                            Sse.AndNot(selector, ifFalse)
-                        );
+                return AdvSimd.FusedMultiplyAdd(a, AdvSimd.Subtract(b, a), t);
             }
-            else if (AdvSimd.IsSupported)
+            else if (Fma.IsSupported)
             {
-                return AdvSimd.BitwiseSelect(selector.AsByte(), ifTrue.AsByte(), ifFalse.AsByte()).As<byte, float>();
+                return Fma.MultiplyAdd(Sse.Subtract(b, a), t, a);
+            }
+            else if (Sse.IsSupported)
+            {
+                return Sse.Add(Sse.Multiply(a, Sse.Subtract(Vector128.Create(1.0f), t)), Sse.Multiply(b, t));
+            }
+            else
+            {
+                // Redundant test so we won't prejit remainder of this method on platforms without AdvSimd.
+                throw new PlatformNotSupportedException();
             }
-
-            return default;
         }
 
-        public static Vector128<double> ConditionalSelectBitwise(Vector128<double> selector, Vector128<double> ifTrue, Vector128<double> ifFalse)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool NotEqual(Vector128<float> vector1, Vector128<float> vector2)
         {
-            Debug.Assert(Sse.IsSupported || AdvSimd.IsSupported);
+            // This implementation is based on the DirectX Math Library XMVector4NotEqual method
+            // https://github.com/microsoft/DirectXMath/blob/master/Inc/DirectXMathVector.inl
 
-            if (Sse2.IsSupported)
+            if (AdvSimd.IsSupported)
             {
-                return Sse2.Or(
-                            Sse2.And(ifTrue, selector),
-                            Sse2.AndNot(selector, ifFalse)
-                        );
+                Vector128<uint> vResult = AdvSimd.CompareEqual(vector1, vector2).AsUInt32();
+
+                Vector64<byte> vResult0 = vResult.GetLower().AsByte();
+                Vector64<byte> vResult1 = vResult.GetUpper().AsByte();
```

Comment:
Same as for `Equal` - could this be done the following way?

```c#
Vector128<uint> vResult = AdvSimd.CompareEqual(vector1, vector2).AsUInt32();
return (AdvSimd.MinPairwise(vResult, vResult).AsUInt64().ToScalar() != 0xFFFFFFFFFFFFFFFFUL);
```

Reply:
Wondering, why this can't be just `return !Equal(vector1, vector2)` ?

Reply:
NaN